### ISSUE TITLE
promxy & victoriametrics wierdness..

### ISSUE THREAD

**Issue by ThomasADavis** [VictoriaMetrics/VictoriaMetrics#20]
Promxy returns a single metric, instead of all the returned metrics, when running against a victoriametrics server.  Not sure where the problem is at..

Prometheus is v2.7.1,
promxy is v0.0.33
victoriametrics is v1.13.0-single-image-0-g8b4babcb

Basically 3 servers - prometheus remote_write to victoriametrics,  two seperate instances of promxy, one against prometheus, one against victoriametrics..

Good query, against prometheus server:

```
[root@dae-vmetric-1 ~]# curl -XGET "http://dae-prom-1:9090/api/v1/query_range?query=node_load1%7Bnode%3D%22d8-r12-c1-n1%22%7D&start=1555352355&end=1555352655&step=15" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   630  100   630    0     0   100k      0 --:--:-- --:--:-- --:--:--  123k
{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "node_load1",
          "alias": "prometheus-node",
          "instance": "192.168.84.113:9100",
          "job": "prometheus-node",
          "node": "d8-r12-c1-n1"
        },
        "values": [
          [
            1555352355,
            "1.17"
          ],
          [
            1555352370,
            "1.53"
          ],
          [
            1555352385,
            "1.31"
          ],
          [
            1555352400,
            "1.11"
          ],
          [
            1555352415,
            "1"
          ],
          [
            1555352430,
            "0.85"
          ],
          [
            1555352445,
            "1.1"
          ],
          [
            1555352460,
            "1.52"
          ],
          [
            1555352475,
            "1.55"
          ],
          [
            1555352490,
            "1.63"
          ],
          [
            1555352505,
            "1.45"
          ],
          [
            1555352520,
            "1.23"
          ],
          [
            1555352535,
            "1.14"
          ],
          [
            1555352550,
            "1.41"
          ],
          [
            1555352565,
            "2.91"
          ],
          [
            1555352580,
            "2.61"
          ],
          [
            1555352595,
            "2.39"
          ],
          [
            1555352610,
            "2.1"
          ],
          [
            1555352625,
            "1.58"
          ],
          [
            1555352640,
            "1.56"
          ],
          [
            1555352655,
            "1.27"
          ]
        ]
      }
    ]
  }
}
```

Good query, against victoriametrics server, prometheus server above remote_write to it

```
[root@dae-vmetric-1 ~]# curl -XGET "http://dae-vmetric-1:8428/api/v1/query_range?query=node_load1%7Bnode%3D%22d8-r12-c1-n1%22%7D&start=1555352355&end=1555352655&step=15" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   695  100   695    0     0   100k      0 --:--:-- --:--:-- --:--:--  113k
{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "node_load1",
          "alias": "prometheus-node",
          "instance": "192.168.84.113:9100",
          "job": "prometheus-node",
          "node": "d8-r12-c1-n1",
          "system": "dae"
        },
        "values": [
          [
            1555352355,
            "1.17"
          ],
          [
            1555352370,
            "1.29"
          ],
          [
            1555352385,
            "1.31"
          ],
          [
            1555352400,
            "1.09"
          ],
          [
            1555352415,
            "1"
          ],
          [
            1555352430,
            "0.87"
          ],
          [
            1555352445,
            "1.1"
          ],
          [
            1555352460,
            "1.75"
          ],
          [
            1555352475,
            "1.55"
          ],
          [
            1555352490,
            "1.53"
          ],
          [
            1555352505,
            "1.45"
          ],
          [
            1555352520,
            "1.35"
          ],
          [
            1555352535,
            "1.1400000000000001"
          ],
          [
            1555352550,
            "3.2600000000000002"
          ],
          [
            1555352565,
            "2.91"
          ],
          [
            1555352580,
            "2.36"
          ],
          [
            1555352595,
            "2.39"
          ],
          [
            1555352610,
            "1.78"
          ],
          [
            1555352625,
            "1.58"
          ],
          [
            1555352640,
            "1.32"
          ],
          [
            1555352655,
            "1.32"
          ]
        ]
      }
    ]
  }
}
```

Bad results, promxy to victoriametrics:

```
[root@dae-vmetric-1 ~]# curl -XGET "http://dae-promxy-1:8082/api/v1/query_range?query=node_load1%7Bnode%3D%22d8-r12-c1-n1%22%7D&start=1555352355&end=1555352655&step=15" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   250  100   250    0     0  18059      0 --:--:-- --:--:-- --:--:-- 19230
{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "node_load1",
          "alias": "prometheus-node",
          "instance": "192.168.84.113:9100",
          "job": "prometheus-node",
          "node": "d8-r12-c1-n1",
          "system": "dae"
        },
        "values": [
          [
            1555352655,
            "0.93"
          ]
        ]
      }
    ]
  }
}
```

Good results, promxy against prometheus server

```
[root@dae-vmetric-1 ~]# curl -XGET "http://dae-promxy-1:9090/api/v1/query_range?query=node_load1%7Bnode%3D%22d8-r12-c1-n1%22%7D&start=1555352355&end=1555352655&step=15" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   630  100   630    0     0  65920      0 --:--:-- --:--:-- --:--:-- 70000
{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "node_load1",
          "alias": "prometheus-node",
          "instance": "192.168.84.113:9100",
          "job": "prometheus-node",
          "node": "d8-r12-c1-n1"
        },
        "values": [
          [
            1555352355,
            "1.17"
          ],
          [
            1555352370,
            "1.53"
          ],
          [
            1555352385,
            "1.31"
          ],
          [
            1555352400,
            "1.11"
          ],
          [
            1555352415,
            "1"
          ],
          [
            1555352430,
            "0.85"
          ],
          [
            1555352445,
            "1.1"
          ],
          [
            1555352460,
            "1.52"
          ],
          [
            1555352475,
            "1.55"
          ],
          [
            1555352490,
            "1.63"
          ],
          [
            1555352505,
            "1.45"
          ],
          [
            1555352520,
            "1.23"
          ],
          [
            1555352535,
            "1.14"
          ],
          [
            1555352550,
            "1.41"
          ],
          [
            1555352565,
            "2.91"
          ],
          [
            1555352580,
            "2.61"
          ],
          [
            1555352595,
            "2.39"
          ],
          [
            1555352610,
            "2.1"
          ],
          [
            1555352625,
            "1.58"
          ],
          [
            1555352640,
            "1.56"
          ],
          [
            1555352655,
            "1.27"
          ]
        ]
      }
    ]
  }
}
```

**Comment by valyala** [comment]
This looks weird. Will try reproducing it and returning back with results.

**Comment by valyala** [comment]
@ThomasADavis , the issue must be fixed in [v1.13.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Verify whether it has been fixed. Re-open the bug if the issue wasn't fixed.
